mono/metadata/unity-liveness.c

```diff
@@ -237,6 +276,59 @@ static gboolean mono_add_process_object(MonoObject *object, LivenessState *state
 	return FALSE;
 }
 
+MONO_API void mono_validate_object_pointer (MonoObject *object)
+{
+	if (object) {
+		void *vtable = NULL;
+		MonoClass *klass = NULL;
+		char *name = NULL;
+
+		vtable = *(void**)(object);
+		klass = object->vtable->klass;
+		name = klass->name;
+
+		g_assert(vtable);
+		g_assert(klass);
+		g_assert(name);
+	}
+}
+
+MONO_API void mono_validate_string_pointer(MonoString *string)
+{
+	mono_validate_object_pointer(&string->object);
+}
+
+static gboolean mono_add_and_validate_object(MonoObject *object, LivenessState *state)
+{
+	if (object) {
+		MonoVTable *vtable = NULL;
+		MonoClass *klass = NULL;
+		char *name = NULL;
+		vtable = GET_VTABLE(object);
+		klass = vtable->klass;
+		name = klass->name;
+
+		g_assert(vtable);
+		g_assert(klass);
+		g_assert(name);
+
+		if (!IS_MARKED(object)) {
+			gboolean has_references = vtable->klass->has_references;
+			if (has_references || should_process_value(object, state->filter)) {
+				block_array_push_back(state->all_objects, object, state);
```

Comment:
I don't think we need to add any objects to collection here as we don't use it for anything. We are only validating.

Reply:
Sure I can try without the array adds. I suspect you are correct.

Reply:
I get crashes if I remove these 🤔